src/Parser.php

```diff
@@ -417,9 +431,16 @@ protected function toAscii(string $label)
      */
     protected function isHostLabel($label): bool
     {
+        static $pattern = '(?(DEFINE)
+                (?<unreserved> [a-z0-9\_\~])
+                (?<gen_delims> [:\/?#\[\]@])
+                (?<sub_delims> [!$&\'()*+,;=])
+                (?<label> (?:(?&unreserved)(?:(?&unreserved)|(?&gen_delims)|(?&sub_delims)|\-){0,61})?(?&unreserved))
```

Comment:
There's no length requirement in the RFC for labels as far as I can tell. Actually, there isn't even a definition of label at all, it just refers to the labels of DNS if DNS is used.